modules/root/root_api.rb

```diff
@@ -12,6 +12,27 @@ class Proxy::RootApi < Sinatra::Base
     end
   end
 
+  get "/features/v2" do
+    begin
+      enabled_plugins = ::Proxy::Plugins.instance.select do |p|
+        p[:name] != :foreman_proxy && \
+          p[:state] == :running && \
+          p[:class].ancestors.include?(::Proxy::Plugin)
+      end
+      content_type :json
+      enabled_plugins.sort { |x, y| x[:name] <=> y[:name] }.map do |p|
+        {
+          'name' => p[:name],
+          'capabilities' => p[:capabilities],
```

Comment:
I assume this could be anything you want data structure wise in which case YES please. This essentially what I wanted with (https://github.com/theforeman/smart_proxy_pulp/pull/11) without going full out implementing it. I'd rather have this than the one off solution.

In my case, I am not necessarily wanting to store "capabilities" so much as settings or properties. Do we forsee a conflict with concept naming there?

Reply:
Maybe we need both concepts? In the Pulp case I could see:

Setting: pulp URL, Pulp version
Capability: RPM, File

But that may be unncessary specification and a single "settings" type object could give freedom to specify whatever you want

Reply:
I think both concepts could be good. Perhaps it needs a generic `options` or `data` hash that every module can fill as they see fit?